tests/libvmi/lifecycle.go

```diff
@@ -0,0 +1,100 @@
+package libvmi
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	vmiSet := VMISet{"vmi": vmi}
+	vmiSet.Setup()
+	return vmiSet["vmi"]
+}
+
+func CleanupVMI(vmi *v1.VirtualMachineInstance) {
+	vmiSet := VMISet{"vmi": vmi}
+	vmiSet.Cleanup()
+}
+
+type VMISet map[string]*v1.VirtualMachineInstance
+
+func NewVMISet() VMISet {
+	return map[string]*v1.VirtualMachineInstance{}
+}
+
+func (vmis VMISet) Setup() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
+
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Creating VMI %q", name))
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).NotTo(HaveOccurred(), "VMI %q should be successfully created", name)
+		vmis[name] = vmi
+	}
+
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Waiting until VMI %q gets ready", name))
+		vmis[name] = tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
```

Comment:
This will work only for cirros images, so either we pass a LoggedIn function to `Setup` or we discover whic one to use checking disk image.

Reply:
You are 100% right.

I will add the parameter everywhere. We could hopefully drop it once https://github.com/kubevirt/kubevirt/pull/4106 is done.